src/BenchmarkDotNet/Validators/ConfigCompatibilityValidator.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Linq;
+using System.Collections.Generic;
+using BenchmarkDotNet.Configs;
+
+namespace BenchmarkDotNet.Validators
+{
+    public class ConfigCompatibilityValidator : IValidator
+    {
+        public static readonly ConfigCompatibilityValidator FailOnError = new ConfigCompatibilityValidator();
+
+        public bool TreatsWarningsAsErrors => true;
+
+        public IEnumerable<ValidationError> Validate(ValidationParameters validationParameters)
+        {
+            var benchmarks = validationParameters.Benchmarks;
+
+            var orderers =
+                benchmarks
+                    .Select(benchmark => benchmark.Config)
+                    .Where(config => config.Orderer != Order.DefaultOrderer.Instance)
+                    .Select(config => config.Orderer)
+                    .Distinct();
+
+            if (orderers.Take(2).Count() > 1)
```

Comment:
IMHO the `Take` is not needed and adds an extra complexity

```suggestion
            if (orderers.Count() > 1)
```